swayfmt/src/utils/language/expr/conditional.rs

```diff
@@ -71,6 +71,11 @@ fn get_full_width_line_style(
         |temp_formatter| -> Result<LineStyle, FormatterError> {
             let mut if_expr_str = FormattedCode::new();
             format_if_expr(if_expr, &mut if_expr_str, temp_formatter)?;
+
+            if if_expr_str.contains('\n') {
+                return Ok(LineStyle::Multiline);
+            }
```

Comment:
Intuitively I'd expect that, if the formatter determines the length is small enough for the expr to be inlined, it should remove the newlines. :thinking:

In #3555 it looks like the newlines were correctly removed for the else block and the first newline in the "then" block, but the final newline wasn't formatted correctly.

I guess we can land this as a patch, but perhaps we should open an issue to revisit handling this properly?

Reply:
Let me look into this again, maybe the fix should be somewhere upstream instead

Reply:
So I relooked at it again and the correct place to look at was indeed upstream when formatting statements [here](https://github.com/FuelLabs/sway/pull/3557/files#diff-b8982d881172082dbaa5ac05c3b19c79ebaa8a7800c719169c630b41d28fef68L71) - inline was not handled